src/lib/weather-data.component.ts

```diff
@@ -1,73 +1,431 @@
+            complete: () => {
+                console.info("Download finished");
+                // let user download file
+                this.createFile();
+            },
+        });
+    }
+
+    /**
+     * create file which user can download
+     * creates a function to bind to download button
+     * @this.downloadAvailable Flag to control if download is authorized
+     */
+    createFile(): void {
+        if (!this.downloadAvailable) {
+            console.log("Download of File deactivated!");
+            return;
+        }
+
+        const dataString = JSON.stringify(this.weatherData, null, 2); // Convert object to string
 
+        const blob = new Blob([dataString], { type: "application/json" }); // Specify correct MIME type
+
+        const url = window.URL.createObjectURL(blob);
+        const a = document.createElement("a");
+
+        if (this.station) {
+            a.href = url;
+            a.download = `WISdoM_Weather_Data_${this.station.name}_${this.usedDataType}_${this.usedResolution}`; // Replace with your desired file name
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+            window.URL.revokeObjectURL(url);
+        }
+    }
+
+    //------------------------------------------------------------------------------ DatePicker Functions --------------------------------------------
+
+    /**
+     *handle event emitted by datepicker and extract the start and end date
+     * @param event emitted when both dates are selected in GUI
+     */
+    handleDatePickEvent(event: any) {
+        this.usedFrom = this.formatTimestamp(event.data.startDate);
+        this.usedUntil = this.formatTimestamp(event.data.endDate);
+    }
+
+    //------------------------------------------------------------------------------ Formatting Functions --------------------------------------------
+
+    /**
+     * transform the used time format to readable one
+     * @param timestamp point in time to reformat
+     * @returns a classical timestamp format
+     */
+    formatTimestamp(timestamp: string): string {
+        let date = new Date(timestamp);
+
+        let hours = String(date.getUTCHours()).padStart(2, "0");
+        let minutes = String(date.getUTCMinutes()).padStart(2, "0");
+        let seconds = String(date.getUTCSeconds()).padStart(2, "0");
+        let day = String(date.getUTCDate()).padStart(2, "0");
+        let month = String(date.getUTCMonth() + 1).padStart(2, "0"); // Months are zero-based
+        let year = date.getUTCFullYear();
+
+        //
+        return `${hours}:${minutes}:${seconds} ${day}.${month}.${year}`;
+    }
+
+    /**
+     * Converts the timestamp being selected into seconds for request
+     * Disregards the Timezone because the dwd-service doesnt take that into account
+     * @param timestamp to be converted
+     * @returns time in seconds (unix timestamp)
+     */
+    convertTimestampToUnix(timestamp: string): number {
```

Comment:
a timestamp usually is a unix timestamp. this may lead to confusion.

```suggestion
    convertDateStringToUnix(dateString: string): number {
```